nemo/collections/common/data/lhotse/dataloader.py

```diff
@@ -457,6 +457,13 @@ def get_lhotse_sampler_from_config(config, global_rank, world_size, tokenizer=No
     cuts, use_iterable_dataset = read_cutset_from_config(config)
     use_iterable_dataset = determine_use_iterable_dataset(use_iterable_dataset, config)
 
+    # Duration filtering, same as native NeMo dataloaders.
+    # We can filter after the augmentations because they are applied only when calling load_audio().
```

Comment:
The duration filtering was late in the pipeline exactly for this reason. There's a number of augmentations and transforms that change the duration of the utterance. Filtering as early as here will result in incorrect behavior.

What is the issue you're facing?

Reply:
Mentioned in the README, but currently its from TOkenPerSecond . Where its dividing by duration and that could lead to zerodivision error